src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -166,3 +166,11 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 # For commands with shared impl between webapp and functionapp and has output, we apply type validation to avoid confusions
 register_cli_argument('appservice web show', 'name', arg_type=webapp_name_arg_type, validator=validate_existing_web_app)
 register_cli_argument('functionapp show', 'name', arg_type=name_arg_type, validator=validate_existing_function_app)
+
+register_cli_argument('functionapp', 'name', arg_type=name_arg_type, id_part='name', help="name of the function")
+register_cli_argument('functionapp create', 'plan', options_list=('--plan', '-p'), completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
+                      help="name or resource id of the function app service plan. Use 'appservice plan create' to get one")
+register_cli_argument('functionapp create', 'consumption_plan_location', options_list=('--consumption-plan-location', '-c'),
+                      help="Geo location for functionsapps with consumption plans. Use 'functionapp list-consumption-locations' to see possible location plans")
```

Comment:
Is this value meaning to use the location based built-in consumption plan? If yes, I suggest to name/comment it more explicitly